api_db.php

```diff
@@ -17,19 +17,27 @@
 // Needs package php5-sqlite, e.g.
 //    sudo apt-get install php5-sqlite
 
-$db = new SQLite3('/etc/pihole/pihole-FTL.db');
-if(!$db)
-	die("Cannot access database");
-
-// Long-Term API functions
-// if (isset($_GET['test']))
-// {
-// 	// $results = $db->query('SELECT * FROM QUERIES order by TIMESTAMP ASC LIMIT 2');
-// 	$results = $db->query('SELECT TIMESTAMP,TYPE,DOMAIN,CLIENT,STATUS FROM QUERIES order by TIMESTAMP ASC');
-// 	echo "N=".$result->numColumns;
-// 	while ($row = $results->fetchArray())
-// 		var_dump($row);
-// }
+function SQLite3_connect($trytoreconnect)
+{
+	try {
+		// connect to database
+		$db = new SQLite3('/etc/pihole/pihole-FTL.db');
+	}
+	catch (Exception $exception) {
+		// sqlite3 throws an exception when it is unable to connect, try to reconnect after 3 seconds
+		if($trytoreconnect)
+		{
+			sleep(3);
+			SQLite3_connect(false);
+		}
+	}
+}
+SQLite3_connect(true);
+
+if(empty($db))
```

Comment:
I don't see where `$db` is declared so it fits this scope? Haven't tested yet, so it may work.